Content.Server/Audio/Jukebox/JukeboxSystem.cs

```diff
@@ -56,13 +61,55 @@ private void OnJukeboxPlay(EntityUid uid, JukeboxComponent component, ref Jukebo
             }
 
             component.AudioStream = Audio.PlayPvs(jukeboxProto.Path, uid, AudioParams.Default.WithMaxDistance(10f))?.Entity;
+
+            if (_entManager.TryGetComponent(component.AudioStream, out AudioComponent? audio))
+            {
+                var length = _audio.GetAudioLength(audio.FileName);
+                component.TimeWhenSongEnds = component.Time + length - TimeSpan.FromSeconds(audio.PlaybackPosition);
+            }
```

Comment:
Rather than deriving this audiosystem should have it, or use an event instead please.

Reply:
done